### PR TITLE
feat: Introduce synchronous OpenFgaClient

### PR DESCRIPTION
**Pull request by booniepepper**
## References

Generated by https://github.com/openfga/sdk-generator/pull/225

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: openfga_sdk/sync/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from openfga_sdk.exceptions import ApiAttributeError
+from openfga_sdk.exceptions import ApiKeyError
+from openfga_sdk.exceptions import ApiException
+from openfga_sdk.exceptions import NotFoundException
+from openfga_sdk.exceptions import UnauthorizedException
+from openfga_sdk.exceptions import ForbiddenException
+from openfga_sdk.exceptions import ServiceException
+from openfga_sdk.exceptions import ValidationException
+from openfga_sdk.exceptions import AuthenticationError
+from openfga_sdk.exceptions import RateLimitExceededError
```
Comment (jpadilla) [inline openfga_sdk/sync/__init__.py:35]: @booniepepper can we remove anything here that's not part of the sync module? Leaving just `ApiClient` and `OpenFgaClient`

File: openfga_sdk/sync/credentials.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
+"""
```
Comment (jpadilla) [inline openfga_sdk/sync/credentials.py:1]: I think we can remove this file too.

File: openfga_sdk/sync/client/client.py [hunk 3]
```diff
@@ -0,0 +1,675 @@
+# coding: utf-8
```
Comment (jpadilla) [inline openfga_sdk/sync/client/client.py:1]: Let's make sure this top level `client` has an `__init__.py` in it.

**Review by jpadilla — CHANGES_REQUESTED** [review]

File: openfga_sdk/sync/__init__.py (hunk 1 above)
Comment (booniepepper) [reply openfga_sdk/sync/__init__.py:35]: Can do, this was necessary to make tests "just work" but I can update the tests too

File: openfga_sdk/sync/client/client.py (hunk 3 above)
Comment (booniepepper) [reply openfga_sdk/sync/client/client.py:1]: Will do

File: openfga_sdk/sync/credentials.py (hunk 2 above)
Comment (booniepepper) [reply openfga_sdk/sync/credentials.py:1]: I will remove parts of it. The `Credentials` class has some async logic that needs to be handled https://github.com/openfga/python-sdk/blob/main/openfga_sdk/credentials.py#L206C15-L206C28

Comment (booniepepper) [reply openfga_sdk/sync/credentials.py:1]: Removed what I can in https://github.com/openfga/python-sdk/pull/40/commits/2f313f464e7007d0cd4b4489a33051bba1de1882

File: openfga_sdk/sync/__init__.py (hunk 1 above)
Comment (booniepepper) [reply openfga_sdk/sync/__init__.py]: Removed what I can in https://github.com/openfga/python-sdk/pull/40/commits/2f313f464e7007d0cd4b4489a33051bba1de1882

**Review by jpadilla — APPROVED** [review]

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**